### PR TITLE
Hibernate query span naming

### PR DESCRIPTION
**Pull request by laurit**
Currently the span name for hibernate queries created with `createQuery` and `createSQLQuery` is the query itself. This query can contain high cardinality components and unsanitized info. This pr changes span name for hibernate queries to follow the same pattern as jdbc queries operation + entity/table name.
Additionally tests that were not present for hibernate 4.3 were copied from 4.0. Hibernate 6 tests were moved to a separate directory as there are some api changes due to which most of the tests won't work on hibernate 6.

### CODE REVIEW

**Review by mateuszrzeszutek — APPROVED** [review]

File: instrumentation/hibernate/hibernate-4.3/javaagent/src/hibernate6Test/java/spring/jpa/PersistenceConfig.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  }
+
+  private Properties additionalProperties() {
+    Properties properties = new Properties();
+    properties.setProperty("hibernate.show_sql", "true");
+    properties.setProperty("hibernate.hbm2ddl.auto", "create");
+    properties.setProperty("hibernate.dialect", "org.hibernate.dialect.HSQLDialect");
+    //    properties.setProperty(
+    //        "hibernate.format_sql",
+    // env.getProperty("spring.jpa.properties.hibernate.format_sql"));
```
[the remark this answers is not in the document]
Comment (laurit) [reply instrumentation/hibernate/hibernate-4.3/javaagent/src/hibernate6Test/java/spring/jpa/PersistenceConfig.java]: removed, this was already in the file that I copied

File: instrumentation/hibernate/hibernate-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/hibernate/SessionMethodUtils.java [hunk 2]
```diff
@@ -37,7 +49,32 @@
...
+        () -> {
+          // set operation to default value that is used when sql sanitizer fails to extract
+          // operation name
+          String operation = "Hibernate Query";
+          String queryString = query;
+          // query might be hql or jpql query where select is optional, prepend select so we can use
+          // sql sanitizer to extract entity/table name
+          if (queryString.trim().toUpperCase(Locale.ROOT).startsWith("FROM ")) {
+            queryString = "SELECT * " + query;
+          }
```
[the remark this answers is not in the document]
Comment (laurit) [reply instrumentation/hibernate/hibernate-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/hibernate/SessionMethodUtils.java]: Good idea, done.

**Review by trask — COMMENTED** [review]
> Additionally tests that were not present for hibernate 4.3 were copied from 4.0.

`hibernate-4.0` instrumentation applies to 4.0+ (no upper bound), what do you think about moving the hibernate 6 tests to the `hibernate-4.0` module, and rename the `hibernate-4.3` module to `hibernate-procedure-call-4.3` (and only have tests for hibernate `ProcedureCall` in that module)?

File: instrumentation/hibernate/hibernate-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/hibernate/SessionMethodUtils.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (mateuszrzeszutek) [reply instrumentation/hibernate/hibernate-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/hibernate/SessionMethodUtils.java]: Nice. Can you add some tests to `SqlStatementSanitizerTest`?

**Review by mateuszrzeszutek — APPROVED** [review]

**Review by trask — APPROVED** [review]
👍

**Pull request merged**